### PR TITLE
Remove remaining bazel protocs and most genrules

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR attempts to eliminate all bazel-based generation logic from the repo. It
accomplishes this by adding new go:generate instructions where appropriate. In
certain situations (like the testdata/ directory), no direct replacement is provided.
Those files should be manually-generated when changed and do not need to be re-generated
as part of a build process.

There are two remaining genrules that this PR does not touch:
- the `dep_version` rule from repo root
- an envoy file move rule from the `pilot` subdirectory

I did not feel comfortable completely removing those rules.

### CODE REVIEW

File: bin/mixer_codegen.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ if [ ! -e $ROOT/Gopkg.lock ]; then
   exit 1
 fi
 
-GOGO_VERSION=$(grep -A 4 "gogo/protobuf" $ROOT/Gopkg.lock | grep version | sed -e 's/^[^\"]*\"//g' -e 's/\"//g')
+GOGO_VERSION=$(sed -n '/gogo\/protobuf/,/\[\[projects/p' $ROOT/Gopkg.lock | grep version | sed -e 's/^[^\"]*\"//g' -e 's/\"//g')
```
Comment (ldemailly) [inline bin/mixer_codegen.sh]: hopefully this works the same on mac

Comment (mandarjog) [reply bin/mixer_codegen.sh]: yes, it works, I tested it on Mac.

File: broker/pkg/model/config/store.go [hunk 2]
```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:generate sh -c ./generate_mock.sh
```
Comment (mandarjog) [inline broker/pkg/model/config/store.go]: should this be a go:generate on the test file?
It is not needed by store.go

Comment (douglas-reid) [reply broker/pkg/model/config/store.go]: gomock completely removed from these files altogether now. unnecessary dependency that had leaked across packages.

File: bin/mixer_codegen.sh [hunk 3]
```diff
@@ -14,7 +14,7 @@ if [ ! -e $ROOT/Gopkg.lock ]; then
   exit 1
```
Comment (wattli) [inline bin/mixer_codegen.sh]: Should we rename this file to something like istio_codegen.sh since it's not only for mixer?

Comment (douglas-reid) [reply bin/mixer_codegen.sh]: maybe. but that's a bigger change (touches lots of files). I'd prefer to do that only as part of a larger cleanup once everything is working.

**Pull request merged**